src/queries/index.ts

```diff
@@ -364,14 +366,31 @@ export const useCollateralInfo = () => {
   };
 };
 
+export function useBalnAllocation() {
```

Comment:
I'd like you to use `react-query`. It will reduce and simplify the code. @hetfly